src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs

```diff
@@ -65,7 +65,7 @@ public string Arguments
             set => _arguments = value;
         }
 
-        public Collection<string> ArgumentList => _argumentList ??= new Collection<string>();
+        public Collection<string> ArgumentList => _argumentList ??= new NullNotAllowedCollection<string>();
```

Comment:
Technically this is a breaking change.  Previously it was fine to add nulls if you then never used the ProcessStartInfo.  Now it fails immediately even if you decide not to use the instance.

Reply:
@stephentoub is your comment meant to suggest we shouldn't take this change, or as a note that it a breaking change?

Reply:
This seems like an unnecessarily aggressive change.  What happens today on Windows if null is passed? If it throws from Start as well, I'm fine with us improving the message / stack of the exception,  but I think it should come from Start rather than from the collection.

Reply:
Using `wine` to see what happens on Windows:
```
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at System.PasteArguments.AppendArgument(ValueStringBuilder& stringBuilder, String argument)
   at System.Diagnostics.ProcessStartInfo.AppendArgumentsTo(ValueStringBuilder& stringBuilder)
   at System.Diagnostics.Process.BuildCommandLine(ProcessStartInfo startInfo, ValueStringBuilder& commandLine)
   at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(String fileName, IEnumerable`1 arguments)
   at console.Program.Main(String[] args) in /tmp/console/Program.cs:line 10
```
I'll update the PR to throw on `Start`.

Reply:
ArgumentNullException from Start on all platforms would be better.